### DISCUSSION TITLE
Pixi Textures Are Rendering as Black Boxes if GPU Acceleration is disabled on browsers

### DISCUSSION THREAD

**Discussion by Amansaxena001** [pixijs/pixijs#10723] [Q&A]
### Context 
We have a simple PIXI.js application where `SVG` icons are loaded from a sprite sheet and displayed on the canvas using textures. However, we're encountering an issue: when `Use hardware acceleration when available` is disabled in the browser, SVG icons render as black boxes on Windows machines, and the app fails to render entirely on macOS as its not able to detect suitable renderer.

I've explored trying to programmatically detect this browser settings, but it seems impossible due to browser limitations and privacy concerns. There are ways but those are not definitive. I'm investigating why this occurs and if PIXI.js requires GPU acceleration to function properly then how should we handle situations where this setting is accidentally disabled in the browser?

As I mentioned it crashes on mac if gpu-acceleration is disabled with error message `not able to detect suitable renderer` but why its not falling back to other available renderers?

Could anyone suggest workarounds or solutions to address this issue? Please refer the code sandbox link for your ref ( run on windows machine and see )

Pixi Version - `7.3.1` 
Codesandbox- https://codesandbox.io/p/sandbox/icons-integration-with-pixi-js-forked-p5c4r8?file=%2Fjs%2Fapp.js%3A13%2C15

**Comment by Amansaxena001** [comment]
@bigtimebuddy

**Reply by bigtimebuddy** [reply]
In v7, `pixi.js` is WebGL only. If you want to fallback to canvas, you need to use `pixi.js-legacy` package instead. This legacy bundle includes both the WebGL and Canvas renderers.

**Reply by Amansaxena001** [reply]
on mac the app crashed but on windows the apps works, the below snippet works fine on mac but on windows even after disabling the gpu acceleration it returns `true`

```
function webgl_detect(return_context)
{
    if (!!window.WebGLRenderingContext) {
        var canvas = document.createElement("canvas"),
             names = ["webgl2", "webgl", "experimental-webgl", "moz-webgl", "webkit-3d"],
           context = false;

        for(var i=0;i< names.length;i++) {
            try {
                context = canvas.getContext(names[i]);
                if (context && typeof context.getParameter == "function") {
                    // WebGL is enabled
                    if (return_context) {
                        // return WebGL object if the function's argument is present
                        return {name:names[i], gl:context};
                    }
                    // else, return just true
                    return true;
                }
            } catch(e) {}
        }

        // WebGL is supported, but disabled
        return false;
    }

    // WebGL not supported
    return false;
}

```

Any suggestions how I can determine if gpu acceleration is disabled in browser on mac & windows using pixi or any custom module so that I can handle the UX better?

**Reply by Amansaxena001** [reply]
@bigtimebuddy Any suggestions how I can determine if gpu acceleration is disabled in browser on mac & windows using pixi or any custom module so that I can handle the UX better?
